### PR TITLE
Fix YQ syntax to parse .github/actions.yml

### PR DESCRIPTION
**Pull request by sloshy**
I've noticed that the sbt-microsites build has been broken for about a month or so, and in trying to figure out why I found several different issues including how it was not loading files locally on my machine properly. After fixing that, I noticed that the build was failing for a different reason: jekyll was not installed. There is a CI precondition to install jekyll specifically for sbt-microsites, and it was failing with a syntax error:

Screen Shot 2021-01-20 at 2 49 11 PM

This error is in the PR linked here: https://github.com/47degrees/sbt-microsites/pull/529

After looking into what was going on I found that the `yq` syntax was incorrect, or perhaps out of date, so I fixed it with a version that I've tested works on my machine.

### CODE REVIEW

**Comment by sloshy** [comment]
@alejandrohdezma I've added you as a reviewer, because it looks like you were the one who added this script to begin with. I want to make sure that I am fixing it properly so that I do not inadvertently cause some downstream errors. Thanks in advance!

**Comment by alejandrohdezma** [comment]
Thanks @sloshy! This seems to be because a new version of `yq` (`v4`) has been released which modified how the library operates.

File: workflows/ci.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
       - name: Setup yq
         run: sudo snap install yq
       - name: Run pre-conditions
-        run: test -f .github/actions.yml && eval "$(yq r .github/actions.yml -D "true" pre.ci)" || true
+        run: test -f .github/actions.yml && eval "$(yq -r .pre.ci .github/actions.yml)" || true
```
Comment (alejandrohdezma) [inline workflows/ci.yml:38]: Taking a look at the [`v3 to v4`](https://mikefarah.gitbook.io/yq/upgrading-from-v3#reading-with-default-value) docs, we should use and "evaluate" expression and recover the default value:

```suggestion
        run: test -f .github/actions.yml && eval "$(yq e '.pre.ci // "true"' .github/actions.yml)" || true
```

File: workflows/ci.yml [hunk 2]
```diff
@@ -48,7 +48,7 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.ADMIN_GITHUB_TOKEN }}
       - name: Run post-conditions
-        run: test -f .github/actions.yml && eval "$(yq r .github/actions.yml -D "true" post.ci)" || true
+        run: test -f .github/actions.yml && eval "$(yq -r .post.ci .github/actions.yml)" || true
```
Comment (alejandrohdezma) [inline workflows/ci.yml:51]: Taking a look at the [`v3 to v4`](https://mikefarah.gitbook.io/yq/upgrading-from-v3#reading-with-default-value) docs, we should use and "evaluate" expression and recover the default value:

```suggestion
        run: test -f .github/actions.yml && eval "$(yq e '.post.ci // "true"' .github/actions.yml)" || true
```

File: workflows/docs.yml [hunk 3]
```diff
@@ -31,14 +31,14 @@ jobs:
       - name: Setup yq
         run: sudo snap install yq
       - name: Run pre-conditions
-        run: test -f .github/actions.yml && eval "$(yq r .github/actions.yml -D "true" pre.docs)" || true
+        run: test -f .github/actions.yml && eval "$(yq -r .pre.docs .github/actions.yml)" || true
```
Comment (alejandrohdezma) [inline workflows/docs.yml:34]: Taking a look at the [`v3 to v4`](https://mikefarah.gitbook.io/yq/upgrading-from-v3#reading-with-default-value) docs, we should use and "evaluate" expression and recover the default value:

```suggestion
        run: test -f .github/actions.yml && eval "$(yq e '.pre.docs // "true"' .github/actions.yml)" || true
```

File: workflows/docs.yml [hunk 4]
```diff
@@ -31,14 +31,14 @@ jobs:
...
       - name: Generate documentation
         run: sbt ci-docs
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           DOWNLOAD_INFO_FROM_GITHUB: true
       - name: Run post-conditions
-        run: test -f .github/actions.yml && eval "$(yq r .github/actions.yml -D "true" post.docs)" || true
+        run: test -f .github/actions.yml && eval "$(yq -r .post.docs .github/actions.yml)" || true
```
Comment (alejandrohdezma) [inline workflows/docs.yml:41]: Taking a look at the [`v3 to v4`](https://mikefarah.gitbook.io/yq/upgrading-from-v3#reading-with-default-value) docs, we should use and "evaluate" expression and recover the default value:

```suggestion
        run: test -f .github/actions.yml && eval "$(yq e '.post.docs // "true"' .github/actions.yml)" || true
```

File: workflows/release.yml [hunk 5]
```diff
@@ -30,7 +30,7 @@ jobs:
       - name: Setup yq
         run: sudo snap install yq
       - name: Run pre-conditions
-        run: test -f .github/actions.yml && eval "$(yq r .github/actions.yml -D "true" pre.release)" || true
+        run: test -f .github/actions.yml && eval "$(yq -r .pre.release .github/actions.yml)" || true
```
Comment (alejandrohdezma) [inline workflows/release.yml:33]: Taking a look at the [`v3 to v4`](https://mikefarah.gitbook.io/yq/upgrading-from-v3#reading-with-default-value) docs, we should use and "evaluate" expression and recover the default value:

```suggestion
        run: test -f .github/actions.yml && eval "$(yq e '.pre.release // "true"' .github/actions.yml)" || true
```

**Review by alejandrohdezma — CHANGES_REQUESTED** [review]

**Comment by alejandrohdezma** [comment]
Also the `post.release` will also need fixing

File: workflows/ci.yml (hunk 1 above)
Comment (sloshy) [reply workflows/ci.yml]: Fixed

File: workflows/ci.yml (hunk 2 above)
Comment (sloshy) [reply workflows/ci.yml]: Fixed

File: workflows/docs.yml (hunk 3 above)
Comment (sloshy) [reply workflows/docs.yml]: Fixed

File: workflows/docs.yml (hunk 4 above)
Comment (sloshy) [reply workflows/docs.yml]: Fixed

File: workflows/release.yml (hunk 5 above)
Comment (sloshy) [reply workflows/release.yml]: Fixed

**Comment by sloshy** [comment]
@alejandrohdezma Just went through and applied your suggestions. I also made an interesting discovery: the version of `yq` on my local machine was in Python, not Go, the one that is being used here. They have a slight difference in their syntax (the Python one uses `-e` style instead of `e`). That is highly confusing. But, I was able to test locally that it works, so that is good. Thank you for reviewing and lmk if there is anything else that looks off.

**Review by alejandrohdezma — APPROVED** [review]
Everything looks good to me now :) Thanks!

**Pull request merged**